Adyen/UI/Form/Items/Address/FormFullAddressItem.swift

```diff
@@ -64,76 +62,98 @@ public final class FullFormAddressItem: FormValueItem<AddressInfo, AddressStyle>
-        let apartmentTextItem = FormTextInputItem(style: style.textField)
-        let optionalText = localizedString(.fieldTitleOptional, localizationParameters)
-        let titleText = localizedString(.apartmentSuiteFieldTitle, localizationParameters)
-        let placeholderText = localizedString(.apartmentSuiteFieldPlaceholder, localizationParameters)
-        apartmentTextItem.title = "\(titleText) \(optionalText)"
-        apartmentTextItem.validator = LengthValidator(minimumLength: 0, maximumLength: 70)
-        apartmentTextItem.placeholder = "\(placeholderText) \(optionalText)"
-        apartmentTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "apartmentSuite")
-
-        return apartmentTextItem
-    }()
+        bind(item: item, to: .stateOrProvince)
+        return item
+    }
 
-    internal lazy var cityTextItem: FormTextInputItem = {
-        let cityTextItem = FormTextInputItem(style: style.textField)
-        cityTextItem.title = localizedString(.cityFieldTitle, localizationParameters)
-        cityTextItem.placeholder = localizedString(.cityFieldPlaceholder, localizationParameters)
-        cityTextItem.validator = LengthValidator(minimumLength: 2, maximumLength: 70)
-        cityTextItem.validationFailureMessage = validationMessage
-        cityTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "city")
+    private func createTextItem(for field: AddressFields, from viewModel: AddressViewModel) -> FormItem {
+        let item = FormTextInputItem(style: style.textField)
+        item.validationFailureMessage = validationMessage
+        item.title = localizedString(viewModel.labels[field], localizationParameters)
+        item.placeholder = localizedString(viewModel.placeholder[field], localizationParameters)
 
-        return cityTextItem
-    }()
+        if viewModel.optionalFields.contains(field) {
+            item.title.map { item.title = $0 + " \(optionalMessage)" }
+        } else {
+            item.validator = LengthValidator(minimumLength: 2, maximumLength: 70)
+        }
 
-    internal lazy var provinceTextItem: FormTextInputItem = {
-        let cityTextItem = FormTextInputItem(style: style.textField)
-        cityTextItem.title = localizedString(.provinceOrTerritoryFieldTitle, localizationParameters)
-        cityTextItem.placeholder = localizedString(.provinceOrTerritoryFieldPlaceholder, localizationParameters)
-        cityTextItem.validator = LengthValidator(minimumLength: 2, maximumLength: 50)
-        cityTextItem.validationFailureMessage = validationMessage
-        cityTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "provinceOrTerritory")
+        bind(item: item, to: field)
+        return item
+    }
 
-        return cityTextItem
-    }()
+    private func bind(item: FormRegionPickerItem, to field: AddressFields) {
+        observers[field].map { remove($0) }
```

Comment:
`observers[field].map(remove(_:))` might be possible too, looks a bit cleaner

Reply:
👍